idaes/core/util/utility_minimization.py

```diff
@@ -0,0 +1,621 @@
+    		         0.5*((blk.Tout[i] - blk.T_[p]) + sqrt((blk.Tout[i] - blk.T_[p])**2 + eps)))
+    	                   for i in coolingdict.keys()))
+    blk.heat_above_pinch = Constraint(pinch_streamsdict.keys(),
+                                        rule=rule_heat_above_pinch)
+
+    # Define variable for heat content below the pinch point
+    blk.QAc = Var(pinch_streamsdict.keys(),
+            initialize = PD.initQAc,
+            bounds = (1e-8, None),
+            doc = 'Heat content bellow pinch [J]',
+            units=pyunits.get_units(Q[str(pinch_streams[0])])
+            )
+
+    # Define a constraint to calculate the varaible QAc
+    def rule_heat_below_pinch(blk, p):
+            return (blk.QAc[p] == sum(blk.Theta[i]*(
+        			0.5*((blk.Tout[i] - blk.T_[p] + DTmin) + sqrt((blk.Tout[i] - blk.T_[p] + DTmin)**2 + eps)) -
+        			0.5*((blk.Tin[i] - blk.T_[p] + DTmin) + sqrt((blk.Tin[i] - blk.T_[p] + DTmin)**2 + eps)))
+        		          for i in heatingdict.keys()))
+    blk.heat_below_pinch = Constraint(pinch_streamsdict.keys(),
+                                        rule=rule_heat_below_pinch)
+
+    # Define variable for Heat of hot utility
+    blk.Qs = Var(
+            initialize = PD.initQs,
+            bounds = (1e-8, None),
+            doc = 'Heating utilities [J]',
+            units=pyunits.get_units(Q[str(pinch_streams[0])])
+            )
+
+    # Define a constraint to solve for Qs
+    # Where 1E-6 is added to both sides of the constraint as a scaling factor
+    def rule_heating_utility(blk, p):
+            return (blk.Qs >= (blk.QAc[p] - blk.QAh[p]))
+    blk.heating_utility = Constraint(pinch_streamsdict.keys(),
+                                        rule=rule_heating_utility)
+
+    # Define variable for Heat of cold utility
+    blk.Qw = Var(
+            initialize = PD.initQw,
+            bounds = (1e-8, None),
+            doc = 'Cooling utilities [J]',
+            units=pyunits.get_units(Q[str(pinch_streams[0])])
+            )
+
+    # Define a constraint to solve for Qw
+    # Where 1E-6 is added to both sides of the constraint as a scaling factor
+    def rule_cooling_utility(blk):
+            return blk.Qw == -sum(Q[i] for i in exchangerdict.keys()) + blk.Qs
+    blk.cooling_utility = Constraint(rule=rule_cooling_utility)
+
+def heat_data(blk, heating, cooling, DG_units=pyunits.Mwatt):
+    """
+    Function for generating necesary data for Duran-Grossmann initialization
+    Allows the calculation of reactors
+    Args:
+        blk: flowsheet
+        heating: Equipment that requires Heat
+        cooling: Equipment from wich heat is being removed
+    Returns:
+        Dictionary for the equipment exchanging heat containing:
+            Tin
+            Tout
+            FCp
+            Q
+    """
+    # Temperature Epsilon to add or subsract 1 degree to avoid division over
+    # 0 in equipment not changing temperature
+    espT = 1
```

Comment:
This looks like it should be a global constant. I recall seeing it in the previous function as well.